db/fixtures/ae_datastore/ManageIQ/Cloud/VM/Retirement/StateMachines/Methods.class/__methods__/amazon_pre_retirement.rb

```diff
@@ -4,12 +4,27 @@
 #
 
 vm = $evm.root['vm']
-if vm && vm.attributes['power_state'] == 'on'
-  ems = vm.ext_management_system
+ems = vm.ext_management_system if vm
+
+if vm.nil? || ems.nil?
+  $evm.log('info', "Skipping Amazon pre retirement for Instance:<#{vm.try(:name)}> on EMS:<#{ems.try(:name)}> "\
+              "with instance store type <#{vm.hardware.root_device_type}>")
+  exit MIQ_OK
+end
+
+power_state = vm.power_state
+if power_state == "on"
+  if vm.hardware.root_device_type.blank?
+    $evm.log('error', "Aborting Amazon pre retirement, empty root_device_type."\
+                      "  Instance may have been provisioned outside of CFME <#{vm.name}>")
```

Comment:
@billfitzgerald0120 
CFME is not valid in messages can we change the second line to 
```
"Instance <#{vm.name}> may have been provisioned externally"
```